docs/content/mongocxx-v3/working-with-bson.md

```diff
@@ -35,12 +35,21 @@ which to use is entirely aesthetic.
 ## List builder {#list}
 
 The simplest way to create a BSON document or array is to use the JSON-like
-list builder: 
+list builder. The type must be specified for nested documents and arrays. 
 
 ```c++
-// { "hello": "world" }
-bsoncxx::builder::list::document list_builder = {"hello", "world"};
-bsoncxx::document::value document = list_builder;
+// { "books" : [ 
+//      { "title" : "The Great Gatsby", "author" : "Francis Scott Key Fitzgerald", "year" : 1925 }, 
+//      { "title" : "One Hundred Years of Solitude", "author" : "Gabriel García Márquez", "year" : 1967 } 
+// ] }
+using namespace builder;
+list::document books = {"books",
+    list::array{
+        list::document{{"title", "The Great Gatsby"}, {"author", "Francis Scott Key Fitzgerald"}, {"year", 1925}},
+        list::document{{"title", "One Hundred Years of Solitude"}, {"author", "Gabriel García Márquez"}, {"year", 1967}}
+    }
+};
+bsoncxx::document::value document = books;
```

Comment:
Perhaps add a comment here explaining that extracting the final document can be done either through assignment, or by calling extract() ?  And add an example call to extract() here as well?

Reply:
Additionally, it might be worth noting that the `books` builder is either invalid for reuse or resets the state of `list::builder` (depending on what the expected behavior is).

Reply:
Done. Added more documentation to this. FYI, the above isn't legal anymore but 
```
bsoncxx::document::value document = books.extract();
```
is.

Reply:
Oh good, that seems more straightforward!